library/hawkular_alerts_group_dampening.py

```diff
@@ -0,0 +1,304 @@
+        """ Checks if an update is required to the current dampening
+
+            Returns:
+               True, if an update is required for the dampening, False otherwise
+        """
+        for key in desired_dampening.keys():
+            if desired_dampening[key] != current_dampening.get(key):
+                return True
+        return False
+
+    def update_group_dampening(self, group_id, dampening_id, dampening):
+        """ Updates a group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        dampening_object = hawkular.alerts.Dampening(dampening)
+        try:
+            self.client.update_group_dampening(group_id, dampening_id, dampening_object)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to update dampening. Error: {error}".format(error=e))
+        self.changed = True
+
+    def create_group_dampening(self, group_id, dampening):
+        """ Creates a group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        new_dampening = hawkular.alerts.Dampening(dampening)
+        try:
+            self.client.create_group_dampening(group_id, new_dampening)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to create dampening. Error: {error}".format(error=e))
+        self.changed = True
+
+    def create_or_update_group_dampenings(self, group_id, dampenings):
+        """ Creates or updates group trigger dampening in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        if not self.group_trigger_exist(group_id):
+            self.module.fail(msg="Group trigger (group_id} does not exist ".format(group_id=group_id))
+        messages = []
+        current_dampenings = self.get_group_dampenings(group_id)
+        current_dampenings_by_trigger_mode = {d["trigger_mode"]: d for d in current_dampenings}
+        for dampening in dampenings:
+            current_matching_dampening = current_dampenings_by_trigger_mode.get(dampening)
+            if current_matching_dampening:
+                dampening_id = current_matching_dampening["dampening_id"]
+                if self.update_required(dampenings[dampening], current_matching_dampening):
+                    self.update_group_dampening(group_id, dampening_id, dampenings[dampening])
+                    messages.append("Successfully updated {trigger_mode} dampening {dampening_id}: {dampening}".format(trigger_mode=dampening, dampening_id=dampening_id, dampening=dampenings[dampening]))
+            else:
+                dampenings[dampening].update({"trigger_mode": dampening})
```

Comment:
Even though I don't think it's really necessary here, as the dampening won't be used again, I'm now using `copy()`.
Thanks